connector/doc/guides/bootstrap_connector.rst

```diff
@@ -172,22 +172,20 @@ Environment
 
 We'll often need to create a new environment to work with.
 I propose to create a helper method which build it for us (in
-``connector_coffee/connector.py``::
+``connector_coffee/models/coffee_backend.py``::
 
-    from openerp.addons.connector.connector import Environment
+    from contextlib import contextmanager
+    from odoo.addons.connector.connector import Environment
 
+    class CoffeeBackend(models.Model):
+        _name = 'coffee.backend'
+        # extend this existing model
 
-    def get_environment(session, model_name, backend_id):
-        """ Create an environment to work with. """
-        backend_record = session.env['coffee.backend'].browse(backend_id)
-        env = Environment(backend_record, session, model_name)
-        lang = backend_record.default_lang_id
-        lang_code = lang.code if lang else 'en_US'
-        if lang_code == session.context.get('lang'):
-            return env
-        else:
-            with env.session.change_context(lang=lang_code):
-                return env
+      @contextmanager
+      @api.multi
+      def get_environment(self, model_name):
```

Comment:
And: yes we could add `get_environment` in the abstract `connector.backend`. Yes the `model_name` is required for the `ConnectorEnvironment`.